src/rpcblockchain.cpp

```diff
@@ -33,9 +33,24 @@ double GetDifficulty(const CBlockIndex* blockindex)
             blockindex = chainActive.Tip();
     }
 
-    int nShift = (blockindex->nBits >> 24) & 0xff;
     uint32_t powLimit =
         UintToArith256(Params().GetConsensus().powLimit).GetCompact();;
+    {
+        if (networkDifficulty && Params().GetConsensus().fPowAllowMinDifficultyBlocks)
+        {
+            // Special difficulty rule for testnet:
+            // If a block's timestamp is more than 2*nPowTargetSpacing minutes after
+            // the previous block, then it is permitted to be min-difficulty. So
+            // get the last non-min-difficulty (or at worst the genesis difficulty).
+            auto window = Params().GetConsensus().nPowTargetSpacing*2;
+            while (blockindex->pprev && blockindex->nBits == powLimit &&
+                    blockindex->GetBlockTime() > blockindex->pprev->GetBlockTime() + window) {
```

Comment:
Not that I could figure out nicely, because the logic is constructed differently in `pow.cpp`.